backend/src/main/java/ai/giskard/service/ee/LicenseService.java

```diff
@@ -0,0 +1,156 @@
+package ai.giskard.service.ee;
+
+import ai.giskard.service.FileLocationService;
+import ai.giskard.service.GiskardRuntimeException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.bouncycastle.crypto.params.Ed25519PublicKeyParameters;
+import org.bouncycastle.crypto.signers.Ed25519Signer;
+import org.bouncycastle.util.encoders.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+public class LicenseService {
+
+    private final Logger log = LoggerFactory.getLogger(LicenseService.class);
+
+    private final FileLocationService fileLocationService;
+
+    private static final String SIGNATURE_KEY = "c947f66224d465b50004c327fc831cff672fc07b540b0613d6f661d0e72d455d";
+
+    @Getter
+    @Setter
+    public class License {
+        private String planName;
+        private String planCode;
+        private List<String> features;
+
+        public Map<FeatureFlagService.FeatureFlag, Boolean> getFeatures() {
+            Map<FeatureFlagService.FeatureFlag, Boolean> map = new HashMap<>();
+
+            for (FeatureFlagService.FeatureFlag featureFlag : FeatureFlagService.FeatureFlag.values()) {
+                map.put(featureFlag, false);
+            }
+
+            if (features != null) {
+                for (String feat : features) {
+                    map.put(FeatureFlagService.FeatureFlag.valueOf(feat), true);
+                }
+            }
+
+            return map;
+        }
+
+        public boolean hasFeature(FeatureFlagService.FeatureFlag flag) {
+            return this.getFeatures().get(flag);
+        }
+    }
+
+    private License license;
+
+    public License getCurrentLicense() {
```

Comment:
getCurrentLicense and uploadLicense should be either `synchronized` or rely on double checked locking in order to prevent race condition where 1 request is modifying a license and another one is reading it at the same time